src/Microsoft.Extensions.Configuration.Abstractions/ConfigurationPath.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Utility methods and constants for manipulating Configuration paths
+    /// </summary>
+    public static class ConfigurationPath
+    {
+        public static readonly string KeyDelimiter = ":";
+
+        public static string Combine(params string[] pathSegements)
```

Comment:
I like having the `ConfigurationPath.Combine()` method. It seems we use it a lot and third party code can use it too. On the other hand, I am not sure we are supposed to have implementation in an Abstractions package, i.e. I am not sure that it meets the bar for being here the same way that the delimiter itself does.

cc @lodejard